views.py

```diff
@@ -68,22 +69,26 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
 
     resource_cls = check_resource_type(res.resource_type)
 
+    if federated_path:
+        res_root = os.path.join(federated_path, res_id)
+    else:
+        res_root = res_id
+
+    bag_modified = "false"
+    metadata_dirty = "false"
+    if istorage.exists(res_root):
+        bag_modified = istorage.getAVU(res_root, 'bag_modified')
+        metadata_dirty = istorage.getAVU(res_root, 'metadata_dirty')
+
```

Comment:
Are you suggesting that we should recreate the bag when the user asks for the metadata xml file and not the bag?